editor.planx.uk/src/@planx/components/NumberInput/model.ts

```diff
@@ -52,6 +54,19 @@ export const numberInputValidationSchema = (input: NumberInput) =>
         }
         return value === "0" ? true : Boolean(parseNumber(value));
       },
+    })
+    .test({
+      name: "check for a whole number",
+      message: "Enter a whole number",
+      test: (value: string | undefined) => {
+        if (!value) {
+          return false;
+        }
+        if (!Number.isInteger(Number(value))) {
```

Comment:
This seemed most reliable in terms of validation while being related to assessing a number. I did have a string check in as well but this felt weird for a number check...